File: android/DFM/src/main/java/com/dontflymoney/viewhelper/DialogSelectClickListener.java
package com.dontflymoney.viewhelper;

import org.json.JSONArray;
import org.json.JSONException;

import android.content.DialogInterface;



public abstract class DialogSelectClickListener implements DialogInterface.OnClickListener
{
	private JSONArray list;
	
	public DialogSelectClickListener(JSONArray list)
	{
		this.list = list;
	}
	
	public void onClick(DialogInterface dialog, int which)
	{
		try
		{
			setResult(list.getJSONObject(which).getString("Text"),
					list.getJSONObject(which).getString("Value"));
		}
		catch (JSONException e)
		{
			handleError(e);
		}
	}
	
	public abstract void setResult(String text, String value);
	public abstract void handleError(JSONException exception);

}

File: android/DFM/src/main/kotlin/com/darakeon/dfm/uiHelpers/dialogs/DialogSelectClickListener.kt
package com.darakeon.dfm.uiHelpers.dialogs

import android.content.DialogInterface

import org.json.JSONArray

abstract class DialogSelectClickListener(private val list: JSONArray?) : DialogInterface.OnClickListener {

	override fun onClick(dialog: DialogInterface, which: Int) {
		if (list != null) {
			setResult(
				list.getJSONObject(which).getString("Text"),
				list.getJSONObject(which).getString("Value")
			)
		}
	}

	abstract fun setResult(text: String, value: String)
}